### ISSUE TITLE
`IssueLabelToken` has extremely low contrast in high contrast color modes

### ISSUE THREAD

**Issue by iansan5653** [primer/react#3098]
### Description

The `IssueLabelToken` component looks fine in most color modes, but is broken in high contrast modes. 

For example, the dark gray token from the Storybook looks like this in light mode:

<img width="123" alt="screenshot of dark gray issue label with black text" src="https://user-images.githubusercontent.com/2294248/228925390-498cd71e-e29f-48b0-a355-9afb7c584c62.png">

But in high contrast mode it looks disabled:

<img width="123" alt="screenshot of light gray issue label with light gray text" src="https://user-images.githubusercontent.com/2294248/228925507-d89b7c36-3a62-4f0d-9ccf-40cb05199079.png">

Here is another example, live (!) in Projects:

<img width="268" alt="screenshot of blue and purple labels in Projects appearing washed out and faded" src="https://user-images.githubusercontent.com/2294248/228925658-e6ab70fe-7de0-4a6b-a622-982b6ce87569.png">

### Steps to reproduce

Go to [this story link](https://primer.style/react/storybook/?path=/story/components-token-features--default-issue-label-token&globals=colorScheme:light_high_contrast)

### Version

v35.23.0

### Browser

Chrome
